src/components/DataListSection.tsx

```diff
@@ -80,20 +80,15 @@ export const DataListSection = ({
   if (listData?.length) {
     return (
       <View>
-        {!!sectionTitle && (
-          <SectionHeader onPress={navigate} title={sectionTitle ?? getTitleForQuery(query)} />
+        <SectionHeader onPress={navigate} title={sectionTitle ?? getTitleForQuery(query)} />
+        {!!limit && (
+          <ListComponent
+            data={isRandom ? _shuffle(listData).slice(0, limit) : listData.slice(0, limit)}
+            horizontal={horizontal}
+            navigation={navigation}
+            query={query}
+          />
```

Comment:
do we need to check the `limit` as well already in line 80? what if there is no limit here? `data` could go wrong where `limit` is used, right? so it seems, that `limit` is mandatory.

ok as i checked now, `limit = 3` is set per default. so we do not need any check for `limit` anymore, right?